Fix some analytics issues

jivey opened the pull request:
for: https://github.com/openstax/unified/issues/2132
Fixes a timing bug with analytics firing before the page title has been updated. (based off #1772)

jivey committed:
initial migration to gtag

jivey committed:
translate old params instead of changing all of them for now

also remove the unused trackEvent

jivey committed:
update specs and use eventAction as event name

jivey committed:
simplify set command

jivey committed:
fix type version

jivey committed:
use dimension directly due to issues with map not working

jivey committed:
update snapshot

jivey committed:
update prerender spec

jivey committed:
lint

jivey committed:
Merge branch 'main' into upgrade-ga

jivey committed:
actually fix prerender test

jivey committed:
test more fixes

jivey committed:
fix test, initial page view should not have click event data

jivey committed:
tidy

jivey committed:
always turn off send_page_view for config calls

jivey committed:
Merge branch 'main' into ga-page-view-false

jivey committed:
temporarily add staging properties to dev env

jivey committed:
remove temp test ids

jivey committed:
Merge branch 'ga-page-view-false' into page-view-hook-bug

jivey committed:
ensure page title gets set before calling analytics

The body of the hook was extracted so it can be called directly
with await and it doesn't seem to actually need the action as
a param.

jivey committed:
organize imports

jivey committed:
fix lint

jivey commented on src/app/content/hooks/receiveContent.ts:
```diff
@@ -66,4 +66,8 @@ const hookBody: ActionHookBody<typeof receivePage | typeof locationChange> = (se
+const hookBody: ActionHookBody<typeof receivePage | typeof locationChange> = (services) => {
+  return async() => updateHead(services);
+};
+
```
I'm not certain if there is a better way to accomplish this, I needed to extract the body due to the types expecting the action to be passed, but I need to call it directly in locationChange without having the page info

TomWoodward replied:
where you're calling this here https://github.com/openstax/rex-web/pull/1779/files#diff-5c521d5b56bf1fc8410db4d851fabef38392a7ee950c8d6d314d7a317d8d5336R25 you have the locationChange action passed in on line 17

jivey replied:
Thank you I totally missed that :(

staxly[bot] committed:
Merge branch 'main' into page-view-hook-bug

staxly[bot] committed:
Merge branch 'main' into page-view-hook-bug

staxly[bot] committed:
Merge branch 'main' into page-view-hook-bug

TomWoodward reviewed (changes requested):
looks good, i think you can undo that change to the hook body, but other than that everything looks fine

staxly[bot] committed:
Merge branch 'main' into page-view-hook-bug

jivey committed:
use hook body directly, temporarily add staging test ids

staxly[bot] committed:
Merge branch 'main' into page-view-hook-bug

jivey committed:
do not track highlight events if the card is not active

There is a bug in the existing analytics code that will cause double
"show create" and "show login" events when switching pages after a
legitimate event has fired. This is caused by the props.data being
a falsey value in the return check. This commit adds a second check
to make sure isActive is true before firing the event.

jivey commented:
@TomWoodward Re-requesting one more review after discovering an existing analytic bug with highlight card: https://github.com/openstax/rex-web/pull/1779/commits/d169577d1e0cb0a939e4f5633a4d06d2bdbbc703

For testing what was happening, I did this:

1. Open a page and start to make a highlight
2. Confirm the analytic event fires for "show create" or "show login" depending on auth status (using the debug extension or network logs)
3. Navigate to another page via TOC
4. See the analytic event fires again without doing anything

I think all that is needed here is the isActive check but this also seems like a good target for a future refactor, it feels like it maybe should skip the entire Card render unless it's active 🤔

TomWoodward reviewed (approved):
Looks fine but you have to remove that dev config

jivey committed:
remove test ids

staxly[bot] committed:
Merge branch 'main' into page-view-hook-bug

Dantemss reviewed (approved):
Tom already reviewed but it really wants me to review too so I'll approve :)
